### ISSUE TITLE
Survival mode

### ISSUE THREAD

**Issue by purretao** [Balimbanana/SM-Synergy#15]
Hi Balimbanana,

I would like to play survival mode on Synergy for HL2. I've managed to use the script to install the server but I don't know which commands to use to play? I cannot seem to be able to summon medkits and respawn is enabled.

Thank you so much.

**Comment by purretao** [comment]
Hi sir,
Happy new year!

I've been trying a bit more to make it work and I get the following errors when starting the server:

ContentLoader[SV]: ERROR: Failed to load resource/GameSpecificClassnames.txt!
Error loading cfg/pure_server_minimal.txt
L 01/01/2024 - 18:51:12: [SM] Exception reported: Failed to create native "GetCustomEntList", name is probably already in use
L 01/01/2024 - 18:51:12: [SM] Blaming: synfixesdev.smx
L 01/01/2024 - 18:51:12: [SM] Call stack trace:
L 01/01/2024 - 18:51:12: [SM]   [0] CreateNative
L 01/01/2024 - 18:51:12: [SM]   [1] Line 1699, C:\SteamCMD\steamapps\common\Synergy\synergy\addons\sourcemod\scripting\synfixesdev.sp::AskPluginLoad2
L 01/01/2024 - 18:51:12: [SM] Failed to load plugin "synfixesdev.smx": unexpected error 23 in AskPluginLoad callback.
L 01/01/2024 - 18:51:12: [SM] Unable to load plugin "synsweps.smx": Native "CheckNPCAlly" was not found

Any clues? 

Thank you !

**Comment by Balimbanana** [comment]
Hello, those 2 errors are:
SynFixes is installed as well as SynFixesDev which conflicts on creating the natives. You have to remove: `steamapps\common\Synergy\synergy\addons\sourcemod\plugins\synfixes.smx`

and CheckNPCAlly is part of HealthDisplay, so you will need to have that plugin installed.

SynModes has a config in `steamapps\common\Synergy\synergy\cfg\sourcemod\synmodes.cfg` which contains a couple CVars that you may want to set up for Survival Mode. For the main gamemode, change `sm_gamemodeset "coop"` to `sm_gamemodeset "survival"`

Then you will want to add to the bottom of the config (it is not present by default) `synsweps_spawnwith "weapon_medkit"`

**Comment by purretao** [comment]
Hi Balimbanana,

Thank you so much, it seems to load now. However, the servers crashes because of the medkit, it seems:

```
L 01/02/2024 - 14:03:24: [SM] Exception reported: Memory leak
L 01/02/2024 - 14:03:24: [SM] Blaming: synsweps.smx
L 01/02/2024 - 14:03:24: [SM] Call stack trace:
L 01/02/2024 - 14:03:24: [SM]   [0] CreateDataPack
L 01/02/2024 - 14:03:24: [SM]   [1] Line 2242, C:\SteamCMD\steamapps\common\Synergy\synergy\addons\sourcemod\scripting\synsweps.sp::GetWepAnim
L 01/02/2024 - 14:03:24: [SM]   [2] Line 6354, C:\SteamCMD\steamapps\common\Synergy\synergy\addons\sourcemod\scripting\synsweps.sp::OnPlayerRunCmd
L 01/02/2024 - 14:03:24: [SM] MEMORY LEAK DETECTED IN PLUGIN (file "synsweps.smx")
L 01/02/2024 - 14:03:24: [SM] MEMORY LEAK DETECTED IN PLUGIN (file "synsweps.smx")
L 01/02/2024 - 14:03:24: [SM] Unloading plugin to free 32296 handles.
L 01/02/2024 - 14:03:24: [SM] Unloading plugin to free 32296 handles.
L 01/02/2024 - 14:03:24: [SM] Contact the author(s) of this plugin to correct this error.
L 01/02/2024 - 14:03:40: --------------------------------------------------------------------------
L 01/02/2024 - 14:03:40: --------------------------------------------------------------------------
L 01/02/2024 - 14:03:40: Type   DataPack            |   Count   32288
L 01/02/2024 - 14:03:40: Type   DataPack            |   Count   32288
L 01/02/2024 - 14:03:40: Type   Timer               |   Count   2
L 01/02/2024 - 14:03:41: Type   Timer               |   Count   2
L 01/02/2024 - 14:03:41: Type   CellArray           |   Count   6
L 01/02/2024 - 14:03:41: Type   CellArray           |   Count   6
L 01/02/2024 - 14:03:41: -- Approximately 526034 bytes of memory are in use by (32296) Handles.
```

- The medkit has no skin, it shows an error model instead (using `use weapon_medkit` in the console). I tried installing weapon packs from your script but the script crashes when doing so. It's not a big deal but if you have an easy fix that doesn't require you much work, I'd appreciate it !
Server side shows: `Error reading weapon data file for: weapon_medkit`

- Might be related as it is a model loader, but I'm also getting these errors:
```
CModelLoader::Map_IsValid:  No such map 'maps/ep1_citadel_00.bsp'
CModelLoader::Map_IsValid:  No such map 'maps/ep1_citadel_01.bsp'
```
HL2 episodes and Lost Coast are installed on my Steam, so I'm not sure why it's giving these errors. I can switch to one of these maps from the server so they are there. 
If it's unrelated to the skin issue, then no problem. 

Thanks again for the help! If I manage to make everything work properly, I will try to make a guide on how to make it run for more people to enjoy it :)

**Comment by Balimbanana** [comment]
Ah alright, the ragdoll removal issue is from `synfixes_remove_playerragdolls` which should be in `steamapps\common\Synergy\synergy\cfg\sourcemod\synfixes.cfg`
The reason it is 1 by default is because of a crash that can occur on creation of player ragdolls, but it generally only happens on higher player counts, so it should be fine to turn it off.

The datapack overflow should be fixed if you update the script. The problem is coming from not having the custom weapon pack, the files are in this repo called: `OldSynergyWeapons.7z` For the server, extract just the folder inside `OldSynergyWeapons.7z/steamapps\common\Synergy\synergy\custom\oldsyn\` and merge the materials/models/scripts/sound directories with your `steamapps\common\Synergy\synergy` SynSweps should add the files to downloads, so not everyone will necessarily need to download it separately.

`CModelLoader::Map_IsValid` messages are just a bug in Synergy, if you want to remove those messages, you can remove all the maps except one from maplist.txt and mapcycle.txt in `steamapps\common\Synergy\synergy`, it can crash if you leave it blank entirely, so just leave something like d1_trainstation_01 in it.
Basically the issue is because the Map_IsValid check will only check for maps that are currently mounted. Synergy will only mount the games/mods that are required to run, so when you are playing HL2, Ep1 and Ep2 are not mounted, but they are selectable.

Spectating in first person is possible with plugins, I don't have one currently, but it is essentially just one property, `m_iObserverMode` set to the right number on dead players.

Currently the custom weapons are still best used on the Regular branch, there have been issues in the Development branch to do with both the weapons and respawn system which I am not entirely sure is fixable at the moment.

**Comment by purretao** [comment]
Hi Balimbanana,

Many thanks for your help, my server is now in a seemingly workable state, waiting for my friends to do a full run of the HL2 games :) The only thing that kinda feels off is the fact that we cannot spectate in First Person (we just played through Black Mesa which had it), but it's pretty minor all things considered.

Just out of curiosity, as I've also downloaded your bhop mod, what's the difference between bhopmode 1 and 2?
In your scripts, did you have any way to customize the difficulty, or should I just adjust skill_mod.cfg for that?

Many thanks again for the huge work you've been doing.
